azerothcore/azerothcore-wotlk
# (Core\Logging\GameEvent) Create Logging and clean up

Add loggin for game_event_creature/game_event_gameobject contains creature/gameobject not found in table creature/gameobject table.
Also Clean Up SQL for reported missing guids per game_event_creature/game_event_gameobject that was reported in the DBerrors.log

With this PR it reports the current errors with AZ-cores DB
![image](https://user-images.githubusercontent.com/16887899/146302061-26d1c0d1-5e98-4659-92f8-ddf8d7817c25.png)

Included SQL to remove the missing guids from the gameevents that is not present in the creature or gameobject tables.


## Changes Proposed:
-  Creates Logging for game_event_creature/game_event_gameobject
-  If creature\gameobject guid does not exsist but is linked in the game_event_creature/game_event_gameobject it reports it in red in the console and is then logged in the DBErrors.log
- Clean Up SQL to remove guids from the game_event_creature/game_event_gameobject that is currently in core.

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Builds and Compiles without error
- Reports missing guids in DBErrors.log


## How to Test the Changes:


1. Launch World
2. observer if world reports any guids missing in creature\gameobject tables that is listed in the gameevent tables.
3. if any is reported check the DBErrors.log

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
